ocdskingfisherarchive/collection.py

```diff
@@ -60,16 +67,46 @@ def get_size_of_data_folder(self):
             universal_newlines=True,
         )
 
-        self.data_size = int(output.split('\t')[0])
-        return self.data_size
+        self._data_size = int(output.split('\t')[0])
+        return self._data_size
+
+    def get_data_files_exist(self):
+        dir_name = self.config.directory_data + '/' + self.source_id + '/' + \
+                   self.data_version.strftime("%Y%m%d_%H%M%S")
+
+        # It must exist and be a directory
+        if not os.path.exists(dir_name) or not os.path.isdir(dir_name):
+            return False
+
+        # No problems found so ...
+        return True
+
+    def _cache_scrapyd_log_file_info(self):
+        if self._scrapy_log_file_name is not None:
+            return self._scrapy_log_file_name
 
-    def _find_scrapyd_log_file(self):
         dir_to_search = os.path.join(self.config.directory_logs, self.source_id)
+        if not os.path.exists(dir_to_search) or not os.path.isdir(dir_to_search):
+            return
         for filename in os.listdir(dir_to_search):
             if filename.endswith(".log"):
                 slf = ScrapyLogFile(os.path.join(dir_to_search, filename))
                 if slf.does_match_date_version(self.data_version):
-                    return os.path.join(dir_to_search, filename)
+                    self._scrapy_log_file = slf
+                    self._scrapy_log_file_name = os.path.join(dir_to_search, filename)
+                    return
+
+    def is_subset(self):
+        self._cache_scrapyd_log_file_info()
+        return self._scrapy_log_file and self._scrapy_log_file.is_subset()
+
+    def has_errors_count(self):
+        self._cache_scrapyd_log_file_info()
+        return True if self._scrapy_log_file else False
+
+    def get_errors_count(self):
+        self._cache_scrapyd_log_file_info()
```

Comment:
This is very repetitive. Instead, you can do:

```python
@property
def scrapy_log_file(self):
    self._cache_scrapyd_log_file_info()
    return self._scrapy_log_file
```

That way, these methods can just call `self.scrapy_log_file.get_...` instead of calling the cache method first.

You can do the same for `scrapy_log_file_name`.

Reply:
Done